File: SRPTests/TestScripts/CreateTextureMips.py
# Test script-created texture mip generation

from SRPScripting import *
import utils

utils.TestSetting('generateMips', True, globals())

col1 = (0, 0, 0, 1)
col2 = (1, 1, 1, 1)

w = 16
h = 16

# Create the texture.
def getPixel(x, y):
	set = x+1 == y or (w - x - 1) == y
	return col1 if set else col2
	
tex = ri.CreateTexture2D(w, h, Format.R8G8B8A8_UNorm, getPixel, generateMips)

utils.TestTextureLevel(ri, tex, 1)

File: SRPTests/TestScripts/CS/CreateTextureMips.cs
// Test script-created texture mip generation
#load "Utils/utils.cs"
using SRPScripting;
using System.Numerics;

var generateMips = TestSetting<bool>("generateMips", true);

var col1 = new Vector4(0, 0, 0, 1);
var col2 = new Vector4(1, 1, 1, 1);

int w = 16;
int h = 16;

// Create the texture.
Vector4 getPixel(int x, int y)
{
	bool set = x+1 == y || (w - x - 1) == y;
	return set ? col1 : col2;
}
	
var tex = ri.CreateTexture2D(w, h, Format.R8G8B8A8_UNorm, getPixel)
	.WithMips(generateMips);

TestTextureLevel(tex, 1);
